vault/resource_rabbitmq_secret_backend_role.go

```diff
@@ -79,50 +117,29 @@ func rabbitmqSecretBackendRoleWrite(d *schema.ResourceData, meta interface{}) er
 	backend := d.Get("backend").(string)
 	name := d.Get("name").(string)
 	tags := d.Get("tags").(string)
-	vhost := d.Get("vhost").([]interface{})
 
-	log.Printf("[DEBUG] Vhosts as list from ResourceData: %+v", vhost)
-
-	vhosts := make(map[string]interface{}, len(vhost))
-
-	for _, host := range vhost {
-		h := map[string]interface{}{}
-		var id string
-		for k, v := range host.(map[string]interface{}) {
-			if k == "host" {
-				id = v.(string)
-				continue
-			}
-			h[k] = v
-		}
-		vhosts[id] = h
+	vhostsJSON, _, err := expandRabbitmqSecretBackendRoleVhost(d.Get("vhost").([]interface{}), "host")
+	if err != nil {
+		return err
 	}
-
-	log.Printf("[DEBUG] vhosts after munging: %+v", vhosts)
-
-	vhostsJSON, err := json.Marshal(vhosts)
+	vhostTopicsJSON, err := expandRabbitmqSecretBackendRoleVhostTopic(d.Get("vhost_topic").([]interface{}))
 	if err != nil {
-		return fmt.Errorf("error serializing vhosts: %s", err)
+		return err
```

Comment:
Similar comment about a more descriptive error for tracking

Reply:
done